### DISCUSSION TITLE
[V8]  Stop table state from automatically resetting on data edit.

### DISCUSSION THREAD

**Discussion by jamTea** [TanStack/table#4475] [Q&A]
Hi there,

I'm looking to recreate the following setup but in V8:
https://react-table-v7.tanstack.com/docs/faq#how-do-i-stop-my-table-state-from-automatically-resetting-when-my-data-changes

I have an editable table. If the user updates the value of a cell, I want to prevent any column sorting and filtering from automatically being applied to the new data. Is that possible in v8?

I really appreciate any help you can provide.

**Comment by Davheed2** [comment]
I set it up this way and it worked for me

const [data, setData] = React.useState([])
const skipPageResetRef = React.useRef()

const updateData = newData => {
   // When data gets updated with this function, set a flag
   // to disable all of the auto resetting
   skipPageResetRef.current = true

   setData(newData)
}

React.useEffect(() => {
   // After the table has updated, always remove the flag
   skipPageResetRef.current = false
})

useReactTable({
   ...
   autoResetPageIndex: !skipPageResetRef.current,
   autoResetExpanded: !skipPageResetRef.current,
})
